tests/util/test_recursive_replace.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import copy
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Union
+
+import pytest
+
+from chia.util.recursive_replace import recursive_replace
+
+
+class TestC:
+    a: int
+    b: str
+
+    def __init__(self, a: int, b: str):
+        self.a = a
+        self.b = b
+
+    def replace(self, **kwargs: Dict[str, Union[int, str, Optional[TestA]]]) -> TestC:
+        ret = TestC(copy.deepcopy(self.a), copy.deepcopy(self.b))
+        for key, value in kwargs.items():
+            if key == "a":
+                ret.a = value  # type: ignore[assignment]
+            elif key == "b":  # pragma: no cover
+                ret.b = value  # type: ignore[assignment]
+            else:  # pragma: no cover
+                raise TypeError(f"unknown field {key}")
+        return ret
+
+
+@dataclass
+class TestA:
+    a: int
+    b: str
+    c: List[int]
+    d: Optional[TestC]
+
+
+class TestB:
+    a: int
+    b: str
+    c: Optional[TestA]
+
+    def __init__(self, a: int, b: str, c: Optional[TestA]):
+        self.a = a
+        self.b = b
+        self.c = c
+
+    def replace(self, **kwargs: Dict[str, Union[int, str, Optional[TestA]]]) -> TestB:
```

Comment:
Since the type is forced to dict and the key is forced to str, you generally just hint the value.  Or maybe you _have_ to just hint the value.

```suggestion
    def replace(self, **kwargs: Union[int, str, Optional[TestA]]) -> TestB:
```

Reply:
oh, I see. I suppose the `Dict` and `str` are implied